tick/survival/convolutional_sccs.py

```diff
@@ -0,0 +1,922 @@
+                proxs = [ProxEquality(0, range=r) for r in prox_ranges]
+            else:
+                raise ValueError("Coeffs are None. " +
+                                 "Equality penalty cannot infer the "
+                                 "coefficients support.")
+        elif n_penalized_features > 0 and self._strength_tv is not None or \
+                self._strength_group_l1 is not None:
+            # TV and GroupLasso penalties
+            blocks_start = np.zeros(n_penalized_features)
+            blocks_end = np.zeros(n_penalized_features)
+            proxs = []
+
+            for i in self.penalized_features:
+                start = int(self._features_offset[i])
+                blocks_start[i] = start
+                end = int(blocks_start[i] + self._n_lags[i] + 1)
+                blocks_end[i] = end
+                if self._strength_tv is not None:
+                    proxs.append(ProxTV(self._strength_tv, range=(start, end)))
+
+            if self._strength_group_l1 is not None:
+                blocks_size = blocks_end - blocks_start
+                proxs.append(ProxGroupL1(self._strength_group_l1,
+                                         blocks_start.tolist(),
+                                         blocks_size.tolist()))
+        else:
+            # Default prox: does nothing
+            proxs = [ProxZero()]
+
+        prox_obj = ProxMulti(tuple(proxs))
+
+        return prox_obj
+
+    def _detect_support(self, coeffs):
+        """Return the ranges over which consecutive coefficients are equal in
+        case at least two coefficients are equal.
+
+        This method is used to compute the ranges for ProxEquality,
+        to enforce a support corresponding to the support of `coeffs`.
+
+         example:
+         coeffs = np.array([ 1.  2.  2.  1.  1.])
+         self._detect_support(coeffs)
+         >>> [(1, 3), (3, 5)]
+         """
+        kernel = np.array([1, -1])
+        groups = []
+        for i in self.penalized_features:
+            idx = int(self._features_offset[i])
+            n_lags = int(self._n_lags[i])
+            if n_lags > 0:
+                acc = 1
+                for change in np.convolve(coeffs[idx:idx+n_lags+1], kernel,
+                                          'valid'):
+                    if change:
+                        if acc > 1:
+                            groups.append((idx, idx + acc))
+                        idx += acc
+                        acc = 1
+                    else:
+                        acc += 1
+                # Last coeff always count as a change
+                if acc > 1:
+                    groups.append((idx, idx + acc))
+        return groups
+
+    @staticmethod
+    def _construct_solver_obj(step, max_iter, tol, print_every,
+                              record_every, verbose, seed):
+        # seed cannot be None in SVRG
+        solver_obj = SVRG(step=step, max_iter=max_iter, tol=tol,
```

Comment:
can you add a comment : # TODO: we might want to use SAGA also later... (might be faster here)

Reply:
ok